addons/event/controllers/main.py

```diff
@@ -23,3 +26,58 @@ def event_ics_file(self, event, **kwargs):
             ('Content-Length', len(content)),
             ('Content-Disposition', content_disposition('%s.ics' % event.name))
         ])
+
+    @route(['/event/my_tickets/<int:event_id>'], type='http', auth='public')
+    def event_my_tickets(self, event_id, registration_ids, tickets_hash, badge_mode=False, simplified=False):
+        """ Returns a pdf response, containing all tickets for attendees in registration_ids for event_id.
+
+        Throw Forbidden if no registration is valid / hash is invalid / parameters are missing.
+        This route is used in links in emails to attendees, as well as in registration confirmation screens.
+
+        :param event: the id of prompted event. Only its attendees will be considered.
+        :param registration_ids: ids of event.registrations of which tickets are generated
+        :param tickets_hash: string hash used to access the tickets.
+        :param badge_mode: boolean, True to use template of foldable badge instead of full page ticket.
+        :param simplified: boolean, True if we want to see the a simplified ticket in a size-adaptative html page.
+        """
+        registration_ids = json.loads(registration_ids or '[]')
+        if not event_id or not tickets_hash or not registration_ids:
+            raise Forbidden(_("These parameters to access tickets are invalid."))
+
+        # We sudo the event in case of invitations sent before publishing it.
+        event_sudo = request.env['event.event'].browse(event_id).exists().sudo()
+        hash_truth = event_sudo._get_my_tickets_access_hash(registration_ids)
+        if not consteq(tickets_hash, hash_truth):
+            raise Forbidden(_("These parameters to access tickets are invalid."))
+
+        event_registrations_sudo = request.env['event.registration'].sudo().search([
+            ('id', 'in', registration_ids),
+            ('event_id', '=', event_sudo.id),
+        ])
+        if not event_registrations_sudo:
+            raise NotFound(_("Those registrations do not exist or are invalid."))
+
+        report_name_prefix = "Ticket" if simplified else "Badges" if badge_mode else "Tickets"
+        report_name = f"{report_name_prefix} - {event_sudo.name} ({event_sudo.date_begin_located})"
+        if len(event_registrations_sudo) == 1:
+            report_name += f" - {event_registrations_sudo[0].name}"
+
+        # sudo is necessary for accesses in templates.
+        if simplified:
+            html = request.env['ir.actions.report'].sudo()._render_qweb_html(
+                'event.action_report_event_registration_simplified_ticket',
+                event_registrations_sudo.ids
```

Comment:
(picky std: better git history if someone add a line after)
```suggestion
                event_registrations_sudo.ids,